python/lsst/faro/measurement/TractMeasurementTasks.py

```diff
@@ -51,15 +51,14 @@ def run(
         self, metricName, catalogs, photoCalibs=None, astromCalibs=None, dataIds=None
     ):
         self.log.info("Measuring %s", metricName)
-        bands = set([f["band"] for f in dataIds])
 
-        if ("g" in bands) & ("r" in bands) & ("i" in bands):
-            rgicatAll = makeMatchedPhotom(dataIds, catalogs, photoCalibs)
+        if ("g" in catalogs) & ("r" in catalogs) & ("i" in catalogs):
+            rgicatAll = makeMatchedPhotom(dataIds, {catalogs[b] for b in 'gri'}, photoCalibs)
             magcut = (rgicatAll["base_PsfFlux_mag_r"] < self.config.faint_rmag_cut) & (
                 rgicatAll["base_PsfFlux_mag_r"] > self.config.bright_rmag_cut
             )
             rgicat = rgicatAll[magcut]
-            extVals = extinction_corr(rgicat, bands)
+            extVals = extinction_corr(rgicat, {'g', 'r', 'i'})
```

Comment:
We seem to be using "gri" explicitly in the code in many places. Can we define the set somewhere earlier and use it here and in the two places above? (also `set("gri")` will create the set you want because the string is treated as an iterator).

Reply:
As I understand, the stellar locus metric here is intended for a specific combination of bands. I would suggest to address this issue of the hard-coded band selection as a separate ticket.

Reply:
so this "gri" just happens to be identical to the "gri" above by chance and not because they are the same gri?